### PR TITLE
Allow the password to be a Sensitive string.

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
For folks doing mass casting of .*password.* to Sensitive,
this should help keep chrony working as expected.

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

**Comment by alexjfisher** [comment]
I guess we already use Sensitive here. https://github.com/voxpupuli/puppet-chrony/blob/f53ea01da517b50d58ff0e6446bcfe6b32df12a2/manifests/config.pp#L27
but this means that we won't store the password as a parameter of the main class in PuppetDB??

**Comment by jcpunk** [comment]
In theory that should only impact folks who pass in a `Sensitive` value.  Normal strings are still accepted.

**Comment by alexjfisher** [comment]
But only if you're on Puppet 6.24.0 or later.  Before then, `unwrap` would error if called on a plain `String`

**Comment by alexjfisher** [comment]
See https://github.com/puppetlabs/puppet/commit/41991b2dc65fd00f484523b915958bd82f0ff836

File: templates/chrony.keys.epp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <% if $chrony::chrony_password and $chrony::chrony_password != 'unset' { -%>
```
Comment (alexjfisher) [inline templates/chrony.keys.epp:1]: I don't think it needs unwrapping here, but the line does look a little broken?  Given its type,  I don't think `$chrony::chrony_password` can ever be `falsey`?

Optionally, feel free to fix that, (and/or parameterize this template) ;)

File: templates/chrony.keys.epp [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-<%= $chrony::commandkey %> <%= $chrony::chrony_password %>
+<%= $chrony::commandkey %> <%= $chrony::chrony_password.unwrap %>
```
Comment (alexjfisher) [inline templates/chrony.keys.epp:2]: Requires puppet 6.24.0 or later.

**Review by alexjfisher — CHANGES_REQUESTED** [review]
A decision needs to be made about either bumping the minimum puppet version in metadata,json and marking this as breaking, or adding the additional logic so that `unwrap` isn't called if the data type isn't `Sensitive`.

**Comment by jcpunk** [comment]
In theory I've reworked the template to better behavior with older puppet.

I may be missing something on the breaking change part.... as I understand it, unless someone explicitly passes in the argument as `Sensitive`, the behavior is the same as before.

File: templates/chrony.keys.epp (hunk 1 above)
Comment (jcpunk) [reply templates/chrony.keys.epp]: I've reworked the template to be smarter.

File: manifests/config.pp [hunk 3]
```diff
@@ -18,12 +18,18 @@
     ),
   }
 
+  $keys_params = {
+    'chrony_password' => Sensitive($chrony::chrony_password),
```
[the remark this answers is not in the document]
Comment (jcpunk) [reply manifests/config.pp]: I've reworked it per your recommendations.  Does this look better?

[the remark this answers is not in the document]
Comment (jcpunk) [reply manifests/config.pp]: Ah, I always miss something, should be fixed now.

**Pull request merged**